PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ServiceBusReceiveModeCheck.java

```diff
@@ -0,0 +1,244 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiDeclarationStatement;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.PsiExpression;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiReferenceExpression;
+import com.intellij.psi.PsiType;
+import com.intellij.psi.PsiVariable;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class extends the LocalInspectionTool and is used to inspect the usage of Azure SDK ServiceBus & ServiceBusProcessor clients in the code.
+ * It checks if the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1.
+ * If the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1, a problem is registered with the ProblemsHolder.
+ */
+public class ServiceBusReceiveModeCheck extends LocalInspectionTool {
+
+    /**
+     * Build the visitor for the inspection. This visitor will be used to traverse the PSI tree.
+     *
+     * @param holder The holder for the problems found
+     * @param isOnTheFly Whether the inspection is on the fly -- not in use
+     * @return The visitor for the inspection. This is not used anywhere else in the code.
+     */
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new ServiceBusReceiveModeVisitor(holder, isOnTheFly);
+    }
+
+    /**
+     * This class extends the JavaElementVisitor and is used to visit the Java elements in the code.
+     * It checks for the usage of Azure SDK ServiceBusReceiver & ServiceBusProcessor clients and
+     * whether the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1.
+     * If the receive mode is set to PEEK_LOCK and the prefetch count is set to a value greater than 1,
+     * a problem is registered with the ProblemsHolder.
+     */
+    static class ServiceBusReceiveModeVisitor extends JavaElementVisitor {
+
+        // Define the holder for the problems found
+        private final ProblemsHolder holder;
+
+        // Define constants for string literals
+        private static final RuleConfig RULE_CONFIG;
+        private static final String SUGGESTION;
+        private static final List<String> METHODS_TO_CHECK;
```

Comment:
I just realised you store a reference to the RULE_CONFIG. You don't need these extra variables `SUGGEESTION`, `METHODS_TO_CHECK`, nor `CLIENTS_TO_CHECK`.  you can use RULE_CONFIG.getMethodsToCheck() when actually using the method.

Reply:
With that being said, you can centralise the logic for SKIP_WHOLE_RULE by adding a method to the RuleConfig class. So you don't have to calculate this variable every time you create a checker.

```java
public class RuleConfig {
  // existing

  public boolean skipRule() {
     return this == Rule.EMPTY || methodsToCheck.isEmpty();
  }
}
```